src/users/InvitationModal.tsx

```diff
@@ -0,0 +1,142 @@
+        props.setInvitations(newInvitationList);
+        if (newInvitationList.length === 0) props.handleClose(); // close the modal if there is no pending invitation left
+      } else {
+        props.addNotification('An error occurred! Please try again!', 2000);
+        console.error('Error cancelling the following pending invitation: ', row, res);
+      }
+    }).catch(console.error);
+  };
+
+  const columnDefinitions = [
+    {
+      titleRenderFunction: () => 'Email',
+      renderFunction: (row: any) => (
+        <span
+          className={tableStyles.CellEllipsis}
+          title={row.email}
+        >
+          {row.email}
+        </span>
+      ),
+      orderingField: null
+    },
+    {
+      titleRenderFunction: () => 'Expires in',
+      renderFunction: (row: any) => {
+        const createdDate = new Date(row.created_at).getTime();
+        const currentDate = new Date().getTime();
+        const numberOfDaysCreated = Math.floor((currentDate - createdDate) / (1000 * 60 * 60 * 24));
+        const numberOfDaysToBeCleanedUp = 15; // pending invitations which are 15 days old will be cleaned up from the Lizard store
+        const numberOfDaysToBeExpired = (numberOfDaysToBeCleanedUp - numberOfDaysCreated) >= 0 ? (numberOfDaysToBeCleanedUp - numberOfDaysCreated) : 0;
+        return (
+          <span
+            className={tableStyles.CellEllipsis}
+          >
+            {numberOfDaysToBeExpired} {numberOfDaysToBeExpired > 1 ? 'days' : 'day'}
+          </span>
+      )},
+      orderingField: null
+    },
+    {
+      titleRenderFunction: () => '', /* Actions */
+      renderFunction: (row: any, tableData:any) => {
+        return (
+            <TableActionButtons
+              tableRow={row}
+              tableData={tableData}
+              setTableData={() => null}
+              triggerReloadWithCurrentPage={() => null}
+              triggerReloadWithBasePage={() => null}
+              actions={[
+                {
+                  displayValue: "Cancel",
+                  actionFunction: (row: any) => deleteAction(row)
+                },
+              ]}
+            />
+        );
+      },
+      orderingField: null
+    }
+  ];
+
+  return (
+    <ModalBackground
+      title={'Pending Invitations'}
+      handleClose={props.handleClose}
+      width={'50%'}
+    >
+      <div
+        style={{
+          padding: 40,
+          paddingBottom: 20,
+          height: '90%',
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-between'
+        }}
+      >
+        <Table
+          tableData={invitations}
+          setTableData={() => null}
+          gridTemplateColumns={'60fr 30fr 10fr'}
+          columnDefinitions={columnDefinitions}
+          dataRetrievalState={dataRetrievalState}
+          triggerReloadWithCurrentPage={() => null}
+          triggerReloadWithBasePage={()=> null}
+          responsive
```

Comment:
An alternative solution would be to use here also the TableStateContainer instead of making the user-table manage this state. Why did you choose here to only use the Table component?

Reply:
Indeed, I didn't want to use the TableStateContainer at first because the table here doesn't contain any search bar, new item button and most importantly no pagination. But think about it again, I think it's better to just use the TableStateContainer, so I changed it accordingly.